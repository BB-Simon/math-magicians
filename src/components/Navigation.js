import { Link } from 'react-router-dom';

const Navigation = () => (
  <nav className="nav">
    nav
  </nav>
);

export default Navigation;
