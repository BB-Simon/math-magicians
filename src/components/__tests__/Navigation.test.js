import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Navigation from '../Navigation';

describe('Jest Snapshot testing suite', () => {
  it('renders correctly', () => {
    const domTree = TestRenderer.create(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    ).toJSON();
    expect(domTree).toMatchSnapshot();

    render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );
    const logoEle = screen.getByText(/Math Magicians/i);
    expect(logoEle).toBeInTheDocument();
    const homeLink = screen.getByText(/Home/i);
    expect(homeLink).toBeInTheDocument();
    const calculatorLink = screen.getByText(/Calculator/i);
    expect(calculatorLink).toBeInTheDocument();
    const quoteLink = screen.getByText(/Quote/i);
    expect(quoteLink).toBeInTheDocument();
  });
});
