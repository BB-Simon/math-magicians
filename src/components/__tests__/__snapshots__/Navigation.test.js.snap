// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite renders correctly 1`] = `
<nav
  className="nav"
>
  <a
    className="link"
    href="/"
    onClick={[Function]}
  >
    Math Magicians
  </a>
  <ul
    className="nav-item"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
